src/components/icon/icon.spec.tsx

```diff
@@ -61,6 +62,35 @@ describe("Icon component", () => {
     { prop: "messages", rendersAs: "message" },
   ];
 
+  describe("console warnings", () => {
+    let loggerSpy: jest.SpyInstance;
+
+    afterEach(() => {
+      loggerSpy.mockRestore();
+    });
+
+    it("should display 'extra-small' deprecation warning once", () => {
+      loggerSpy = jest.spyOn(Logger, "deprecate");
+
+      mount(<Icon type="home" bgSize="extra-small" />);
+
+      expect(loggerSpy).toHaveBeenCalledWith(
+        "The `extra-small` variant of the `bgSize` prop for `Icon` component has been deprecated and will soon be removed."
+      );
+      expect(loggerSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it("should display larger bgSize warning once", () => {
+      loggerSpy = jest.spyOn(console, "warn");
```

Comment:
**suggestion** (non-blocking): I don't think we should call the mock `loggerSpy` in this case as it's not for anything to do with the `Logger`, it's for `console.warn`. I'd call it something like `warnSpy` instead. Obviously that's a bit less convenient because you'd need two separate variables and to `mockRestore` on both - so don't feel obliged to do it.